includes/ucf-events-common.php

```diff
@@ -130,6 +130,105 @@ function ucf_events_display_classic_after( $items, $title ) {
 
 }
 
+if ( !function_exists( 'ucf_events_display_modern_before' ) ) {
+
+	function ucf_events_display_modern_before( $items, $title, $display_type ) {
+		ob_start();
+	?>
+		<div class="ucf-events ucf-events-modern">
+	<?php
+		echo ob_get_clean();
+	}
+
+	add_action( 'ucf_events_display_modern_before', 'ucf_events_display_modern_before', 10, 3 );
+
+}
+
+if ( !function_exists( 'ucf_events_display_modern_title' ) ) {
+
+	function ucf_events_display_modern_title( $items, $title, $display_type ) {
+		$formatted_title = $title;
+
+		switch ( $display_type ) {
+			case 'widget':
+				// title is already formatted at the widget level
+				break;
+			case 'default':
+			default:
+				$formatted_title = '<h2 class="ucf-events-title h4 text-uppercase mb-4">' . $title . '</h2>';
+				break;
+		}
+
+		echo $formatted_title;
+	}
+
+	add_action( 'ucf_events_display_modern_title', 'ucf_events_display_modern_title', 10, 3 );
+
+}
+
+if ( !function_exists( 'ucf_events_display_modern' ) ) {
+
+	function ucf_events_display_modern( $items, $title ) {
+		if ( ! is_array( $items ) ) { $items = array( $items ); }
+		ob_start();
+	?>
+		<div class="ucf-events-list">
+
+		<?php if ( $items ): ?>
+			<?php
+			$numItems = count( $items );
+			$i = 0;
+			foreach( $items as $event ) :
```

Comment:
You can probably do a `foreach( $items as $i => $event) :` here if you want to avoid having to declare a separate iterator. Also this should ensure the `$i` variable is reset on each loop I believe.